romanesco/tasks/workflow.py

```diff
@@ -0,0 +1,123 @@
+import romanesco
+
+from romanesco.utils import toposort
+
+
+def run(task, inputs, outputs, task_inputs, task_outputs):
```

Comment:
I just passed all the vars that were referenced by the internals of each mode, we can refactor the implementations so we don't have to pass all of these to the functions.